Dataset.py

```diff
@@ -261,6 +270,9 @@ def is_cached(self, start, end):
 
   def get_seq_length_nd(self, sorted_seq_idx):
     """
+    TODO: This function is only replaced in CachedDataset, and used only in get_seq_lenght of the
+      Dataset itself, so this code should never be executed.
```

Comment:
It is executed all the time. See the usages. This is called in `get_seq_length`.
I agree that this function (or API) is there for historical reasons. But you cannot just remove it. First you would need to fix all usages. Basically you would need to implement `CachedDataset.get_seq_length` instead. (If I'm not missing something now. This has to be tested well, as this is very critical code.)